lib/plugins/threaded-signing/plugin.c

```diff
@@ -188,59 +240,61 @@ threaded_openssl_sign_hash(sv_threaded_plugin_t *self, const signature_info_t *s
   SignedVideoReturnCode status = SV_UNKNOWN_FAILURE;
   g_mutex_lock(&self->mutex);
 
-  // If the signature has not yet been pulled, or even generated, a new signature cannot be
-  // generated without replacing it. Log in |nbr_of_unsigned_hashes| and move to done.
-  if (self->plugin_state != THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN) {
-    self->nbr_of_unsigned_hashes++;
-    status = SV_OK;
-    goto done;
-  }
-
   // If no |self->signature_info| exists. Allocate necessary memory for it and copy the
   // |private_key| and |algo|.
   if (!self->signature_info) {
     self->signature_info = local_signature_info_create(signature_info);
-    if (!self->signature_info) goto catch_error;
+    // Failed in memory allocation.
```

Comment:
I think this comment should go _inside_ the if-statement for correct interpretation. Think of this as reading a book. Line 246 says ```signature_info``` is created. Then I read we failed allocating memory.

There are more places.

Reply:
I second this. Generally I like to have comments just like docstrings in python, the first thing you read after you read the function name and parameters. Meaning that after I've read the context (function signature) I want to read any other info about that is useful.

Of course if the comment is "Check something..." then it should be before the statement. Example:

// Check if we exceed the max gop and do something about it
if (gop > max_gop) {
  do_something();
}

But generally it's better to have the statement after a check, since usually the check itself it clear:

if (gop > max_gop) {
  // When exceeding the max gop we need to do something
  do_something();
}

It makes it look cleaner and easier to read IMHO.